authenticators/components/org.wso2.carbon.identity.sample.oauth2.federated.authenticator/src/main/java/org/wso2/carbon/identity/sample/oauth2/federated/authenticator/OAuth2CustomAuthenticator.java

```diff
@@ -0,0 +1,108 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.wso2.carbon.identity.sample.oauth2.federated.authenticator.OAuth2CustomAuthenticatorConstants.*;
+
+/***
+ * OAuth2 Custom Authenticator is an outbound authenticator
+ * This extends OAuth2 Generic Authenticator implementation
+ */
+public class OAuth2CustomAuthenticator extends Oauth2GenericAuthenticator {
+
+    private static final long serialVersionUID = 6614257960044886319L;
+
+    @Override
+    public String getFriendlyName() {
+
+        return AUTHENTICATOR_FRIENDLY_NAME;
+    }
+
+    @Override
+    public String getName() {
+
+        return AUTHENTICATOR_NAME;
+    }
+
+    // TODO: Override buildClaims to match the response of respective OAuth2 identity provider
+
+    @Override
+    public List<Property> getConfigurationProperties() {
+
+        List<Property> configProperties = new ArrayList<>();
+
+        Property clientId = new Property();
+        clientId.setName(CLIENT_ID);
+        clientId.setDisplayName(CLIENT_ID_DP);
+        clientId.setRequired(true);
+        clientId.setDescription(CLIENT_ID_DESC);
+        clientId.setDisplayOrder(1);
+        configProperties.add(clientId);
+
+        Property clientSecret = new Property();
+        clientSecret.setName(CLIENT_SECRET);
+        clientSecret.setDisplayName(CLIENT_SECRET_DP);
+        clientSecret.setRequired(true);
+        clientSecret.setConfidential(true);
+        clientSecret.setDescription(CLIENT_SECRET_DESC);
+        clientSecret.setDisplayOrder(2);
+        configProperties.add(clientSecret);
+
+        Property callbackUrl = new Property();
+        callbackUrl.setName(CALLBACK_URL);
+        callbackUrl.setDisplayName(CALLBACK_URL_DP);
+        callbackUrl.setRequired(true);
+        callbackUrl.setDescription(CALLBACK_URL_DESC);
+        callbackUrl.setDisplayOrder(3);
+        configProperties.add(callbackUrl);
+
+        Property authorizationUrl = new Property();
+        authorizationUrl.setName(AUTHZ_URL);
+        authorizationUrl.setDisplayName(AUTHZ_URL_DP);
+        authorizationUrl.setRequired(true);
+        authorizationUrl.setDescription(AUTHZ_URL_DESC);
+        authorizationUrl.setDisplayOrder(4);
+        configProperties.add(authorizationUrl);
+
+        Property tokenUrl = new Property();
+        tokenUrl.setName(TOKEN_URL);
+        tokenUrl.setDisplayName(TOKEN_URL_DP);
+        tokenUrl.setRequired(true);
+        tokenUrl.setDescription(TOKEN_URL_DESC);
+        tokenUrl.setDisplayOrder(5);
+        configProperties.add(tokenUrl);
+
+        Property userInfoUrl = new Property();
+        userInfoUrl.setName(USER_INFO_URL);
+        userInfoUrl.setDisplayName(USER_INFO_URL_DP);
+        userInfoUrl.setRequired(true);
+        userInfoUrl.setDescription(USER_INFO_URL_DESC);
+        userInfoUrl.setDisplayOrder(6);
+        configProperties.add(userInfoUrl);
+
+        return configProperties;
+    }
+}
+
```

Comment:
Remove the extra new line

Reply:
Fix in other files as well

Reply:
Fixed with https://github.com/wso2/samples-is/pull/360